x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -1056,8 +1056,9 @@ const updateExistingDataStream = async ({
   } catch (err) {
     if (
       isResponseError(err) &&
-      err.statusCode === 400 &&
-      err.body?.error?.type === 'illegal_argument_exception'
+      ((err.statusCode === 400 && err.body?.error?.type === 'illegal_argument_exception') ||
+        // handling the mapper_exception error, it should also trigger a rollover
+        err.message.indexOf('mapper_exception') !== -1)
```

Comment:
nit: there is no `error.type` we could use here? it will be more future proof than looking at the message

Reply:
That's the error object:
```
 {
  name: 'ResponseError',
  message: 'mapper_exception\n' +
    '\tRoot causes:\n' +
    "\t\tmapper_exception: the [subobjects] parameter can't be updated for the object mapping [_doc]"
}
```

I'll look at implementing the suggestions [on this comment](https://github.com/elastic/kibana/pull/186991#issuecomment-2191895239) instead.